data/templates/secure_erase.py

```diff
@@ -0,0 +1,341 @@
+    else:
+        exit_status["exit_code"] = exit_code
+    return exit_status
+
+def robust_check_output(cmd, log):
+    '''
+    Subprocess check_output module with try-except to catch CalledProcessError
+    :param cmd: command option for subprocess.check_output, an array
+    :param log: an opened file object to store stdout and stderr
+    :return: a dict include exit_code and command execution message
+    '''
+    assert isinstance(cmd, list), "Input commands is not an array"
+    exit_status = {"exit_code": 0, "message": "check_output command succeeded"}
+    log.write("[" + " ".join(cmd) + "] output:\n")
+    try:
+        output = subprocess.check_output(cmd, shell=False, stderr=log)
+    except subprocess.CalledProcessError as exc:
+        exit_status["message"] = exc.output
+        exit_status["exit_code"] = exc.returncode
+    else:
+        exit_status["message"] = output
+        log.write(str(exit_status) + "\n")
+    return exit_status
+
+def hdparm_check_drive_status(pattern, disk_name, log):
+    '''
+    Verify drive SE status use "hdparm -I /dev/sdx" command.
+    :param patten: re patten to match different SE status
+    :param disk_name: disk device name
+    :param log: an opened file object to store logs
+    '''
+    command = ["hdparm", "-I", disk_name]
+    exit_status = robust_check_output(cmd=command, log=log)
+    assert exit_status["exit_code"] == 0, "Can't get drive %s SE status" % disk_name
+    output = exit_status["message"]
+    secure_index = output.find("Security")
+    assert secure_index != -1, \
+        "Can't find security info, probably disk %s doesn't support SE" % disk_name
+    output_secure_items = output[secure_index:-1]
+    assert pattern.match(output_secure_items), "Disk is not enabled for secure erase"
+    return
+
+def hdparm_secure_erase(disk_name):
+    '''
+    Secure erase using hdparm tool
+    :param disk: an array includes all disk_list will be erased
+    :param arg: secure erase command argument
+    :return: a dict includes SE command exitcode and SE message
+    '''
+    #enhance_se = ARG_LIST.e
+    log_file = disk_name.split("/")[-1] + ".log"  #log file for sdx will be sdx.log
+    log = open(log_file, "a")
+    record_timestamp(log=log, action="start")
+
+    #Hdparm SE Step1: check disk status
+
+    #Secure Erase supported output example
+    #Security:
+    #        Master password revision code = 65534
+    #                supported
+    #        not     enabled
+    #        not     locked
+    #        not     frozen
+    #        not     expired: security count
+    #                supported: enhanced erase
+    #        2min for SECURITY ERASE UNIT. 2min for ENHANCED SECURITY ERASE UNIT.
+    #Checksum: correct
+
+    #except for "supported", other items should have "not" before them
+    #pattern_ese_support = re.compile(r'[\s\S]*supported: enhanced erase[\s\S]*')
+    pattern_se_support = re.compile(r'[\s\S]*(?!not)[\s]*supported[\s]*not[\s]*enabled[\s]*not[\s]'
```

Comment:
Should there some 'space' between 'not' and 'enabled' ? If so, use `+` {1:n} instead of `*` {0:n}